contracts/nft/src/execute/approve_all.rs

```diff
@@ -0,0 +1,25 @@
+use cosmwasm_std::{Addr, DepsMut, Env, MessageInfo, Response};
+use cw_utils::Expiration;
+
+use crate::core::error::ContractError;
+use crate::storage::operators::OPERATORS;
+use crate::util::action::{Action, ActionType};
+
+pub fn handle(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    operator: Addr,
+    expires: Option<Expiration>,
+) -> Result<Response, ContractError> {
+    // reject expired data as invalid
+    let expires = expires.unwrap_or_default();
+    if expires.is_expired(&env.block) {
+        return Err(ContractError::Expired {});
+    }
+
+    // set the operator for us
+    OPERATORS.save(deps.storage, (&info.sender, &operator), &expires)?;
```

Comment:
Do we need any sort of permission check? Is it a problem if anyone can make someone an operator?

Reply:
not for this one since it's the sender is adding permission